tests/contenttypes_tests/test_models.py

```diff
@@ -259,6 +259,22 @@ def test_missing_model(self):
         ct_fetched = ContentType.objects.get_for_id(ct.pk)
         self.assertIsNone(ct_fetched.model_class())
 
+    @mock.patch('django.apps.apps.get_model', return_value=Site)
+    def test_shortcut_view_with_empty_site_fk(self, get_model):
+        """
+        Check that the shortcut view catch an AttributeError raised when the
+        model's ForeignKey to site is equal to None.
+        Refs #26085.
+        """
+        request = HttpRequest()
+        request.META = {
+            "SERVER_NAME": "Example.com",
+            "SERVER_PORT": "80",
+        }
+        object_ct = ContentType.objects.get_for_model(ModelWithEmptyFKToSite)
+        obj = ModelWithEmptyFKToSite.objects.create(title="object title")
+        self.assertRaises(Http404, shortcut, request, object_ct.id, obj.id)
```

Comment:
Use `with self.assertRaisesMessage():` -- this isn't currently working as a correct regression test for the issue because the `Site`model lacks a `get_absolute_url()` method so a different 404 is being raised from the one you're trying to test.